docs/proposals/portable-task-format.md

```diff
@@ -0,0 +1,975 @@
+# Proposal: Portable Task Format for MCP Server Evaluation
+
+## Problem Statement
+
+gevals currently uses bash scripts for setup, verification, and cleanup of evaluation tasks. While this approach works, it creates several problems as the project scales:
+
+**Script complexity grows quickly.** What starts as a simple `kubectl get pod` check becomes 90 lines of bash with jq parsing, retry loops, timeout handling, and error messages. These scripts are hard to read, debug, and maintain. See `examples/kube-mcp-server/tasks/create-canary-deployment/verify.sh` or `examples/kube-mcp-server/tasks/setup-dev-cluster/verify.sh` for examples of this complexity.
+
+**Tasks are not portable across evaluation tools.** An MCP server developer who publishes evaluation tasks today is publishing gevals-specific artifacts. A user who wants to run these same task scenarios against their own application using a different evaluation framework (one focused on response quality rather than MCP behavior) cannot easily reuse them. The task definition is tangled with gevals-specific execution.
+
+**Verification lives in two places.** The current design separates "did the task succeed" (verify script in task.yaml) from "did the agent use MCP correctly" (assertions in eval.yaml). This layering exists for good reason (the same task can run against different MCP servers), but it's confusing for developers writing simple evaluations.
+
+**Language barriers limit contribution.** A Go developer who wants to write verification logic must write bash. A Python developer must write bash. Everyone writes bash, regardless of whether it's the right tool for the job. While shebangs technically allow other languages, the lack of structured I/O makes this awkward.
+
+**Registry automation is fragile.** If an MCP registry wants to periodically run evaluations against registered servers and publish quality scores, it needs to execute arbitrary bash scripts. There's no structured contract for what these scripts need or produce.
+
+## Requirements
+
+Any solution must satisfy these constraints:
+
+### Portability
+
+1. Task definitions must be consumable by tools other than gevals. An evaluation framework focused on response quality should be able to load a task's prompt, run it through their own agent, and apply their own quality metrics.
+
+2. Tasks must be self-contained. All artifacts needed to run a task (manifests, expected files, etc.) should be bundled with or referenced by the task definition.
+
+3. Tasks must not require a specific programming language runtime. A task authored by a Go developer should be runnable on a system that only has Python, and vice versa.
+
+### Expressiveness
```

Comment:
I think this task definition language is a great idea for the reasons mentioned!

Please include "what current scripts express" as a measure of "is it expressive enough". For current scripts I'd point to the ones in: https://github.com/genmcp/gevals/pull/34/files#diff-c7596df0e00fb3e2fd5d79d3291a9f54cbe9c8807ef56a8bb917038ed4d9e544

The longest one of which does weigh in at 200+ lines 🤭